Spigot-API-Patches/0126-PlayerShootProjectileEvent.patch

```diff
@@ -0,0 +1,108 @@
+From d41956f696f449eccd35e72d49b6b7e3edbdc4ac Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Thu, 5 Jul 2018 09:32:54 -0500
+Subject: [PATCH] PlayerShootProjectileEvent
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/event/entity/PlayerShootProjectileEvent.java b/src/main/java/com/destroystokyo/paper/event/entity/PlayerShootProjectileEvent.java
+new file mode 100644
+index 00000000..81a98faa
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/entity/PlayerShootProjectileEvent.java
+@@ -0,0 +1,93 @@
++package com.destroystokyo.paper.event.entity;
++
++import org.bukkit.entity.Entity;
++import org.bukkit.entity.Player;
++import org.bukkit.entity.Projectile;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.bukkit.inventory.ItemStack;
++
++/**
++ * Called when a player shoots a projectile
++ */
++public class PlayerShootProjectileEvent extends EntityEvent implements Cancellable {
++    private static final HandlerList handlers = new HandlerList();
++    private final Entity projectile;
++    private final ItemStack itemStack;
++    private boolean consumeStack = true;
++    private boolean cancelled;
++
++    public PlayerShootProjectileEvent(Player shooter, ItemStack itemStack, Projectile projectile) {
++        super(shooter);
++        this.itemStack = itemStack;
++        this.projectile = projectile;
++    }
++
++    @Override
++    public Player getEntity() {
++        return (Player) entity;
++    }
++
++    /**
++     * Get the shooter of the projectile
++     *
++     * @return The shooter
++     */
++    public Player getShooter() {
```

Comment:
again becomes less useful with getPlayer(), PlayerShootItemEvent is clear enough getPlayer is the shooter, we don't need this method.